### ISSUE TITLE
Missing space characters in GUI labels

### ISSUE THREAD

**Issue by 0x6d696368** [NationalSecurityAgency/ghidra#1185]
**Describe the bug**
Several GUI elements are missing space characters.

**To Reproduce**
1. Get the latest version via git.
2. Open Ghidra
3. GUI labels are missing space characters

**Expected behavior**
The GUI labels have correct space characters between words.

**Screenshots**
![screenshot_Ghidra__NO_ACTIVE_PROJECT_20191029T140812+0100](https://user-images.githubusercontent.com/22860825/67770136-12bcb800-fa56-11e9-8411-c3b732abb307.png)

**Environment (please complete the following information):**
 - OS: CentOS 7
 - Java Version: 11.0.5
 - Ghidra Version: 9.1-DEV ea554996235412571588eca36c4801b71dda087d
 - Ghidra Version: 9.2-DEV 068cac365711c658505ce8f44b1d729e061b7252

It works find in 9.1-BETA downloaded from ghidra-sre.org.

**Comment by ryanmkurtz** [comment]
Did you do a build or are you running out of Eclipse?

**Comment by 0x6d696368** [comment]
> Did you do a build or are you running out of Eclipse?

Build via `gradle buildGhidra`.
I will try running from Eclipse and report back tomorrow.

**Comment by 0x6d696368** [comment]
Running from Eclipse the space characters are present (only tested with ea55499).

Cleaning everything and making a fresh rebuild solved the issue.

I. however. couldn't determine the root cause.

**Comment by ahroach** [comment]
For posterity, let me add to instances where I've seen the same behavior with both Ghidra 9.0.4 and 9.1.

I first encountered this behavior when running Ghidra in Sway, a tiling Wayland window manager. Ghidra refused to start initially, which was related to Sway being a nonreparenting window manager. Running 'wmname LG3D' is a common hack to allow AWT applications to start in nonreparenting window managers. Indeed, this allowed Ghidra to start, but space characters were missing as you noted. Finally I settled on instead setting the _JAVA_AWT_WM_NONREPARENTING environment variable, which allowed Ghidra to start and resulted in text being displayed properly.

I also encountered the same problem when running Ghidra in Gnome under Wayland. Again setting the _JAVA_AWT_WM_NONREPARENTING environment variable resulted in correct text display, though this environment variable was not required for Ghidra to start under Gnome as it was under Sway.

All of this was to say that there seemed to be some mysterious Wayland/AWT interaction at play when I encountered this problem. In both cases I found a workaround, but I was far from understanding the root cause.

**Comment by 0x6d696368** [comment]
I'm now getting this on the 9.1_PUBLIC downloaded from ghidra-sre.org. HELP!

@ahroach thanks for the hint. I am running i1wm. I will investigate further.

**Comment by 0x6d696368** [comment]
Running the 9.1_PUBLIC downloaded from ghidra-sre.org once via Eclipse (from GhidraDev plugin while testing an extension) has fixed the problem, also for running Ghidra 9.1 from outside of Eclipse.

I still have no idea what caused it. I will keep this issue open and either add more information as I find it or close the issue in case the problem doesn't reappear again.

**Comment by ahroach** [comment]
One more note on the Gnome/Wayland scenario: it turns out that the first time I run Ghidra 9.1 under Gnome/Wayland, the text is missing spaces. If I restart Ghidra, the text is correct. And if I delete the "USER_AGREEMENT=ACCEPT" line from .ghidra/.ghidra_9.1_PUBLIC/preferences, the next time I start Ghidra text is again missing spaces, both in the user agreement window and other windows as part of that session.

All of this was in a VM running Debian Stable with openjdk version 11.0.5+10.

**Comment by 0x6d696368** [comment]
> It turns out that the first time I run Ghidra 9.1 under a variety of window manager setups (Gnome/Wayland, Gnome/Xorg, i3/Xorg), the text is missing spaces. If I restart Ghidra, the text is correct. And if I delete the "USER_AGREEMENT=ACCEPT" line from .ghidra/.ghidra_9.1_PUBLIC/preferences, the next time I start Ghidra text is again missing spaces, both in the user agreement window and other windows as part of that session.

Can confirm for my setup i3/X11. When running Ghidra 9.1 and `USER_AGREEMENT=ACCEPT` is not in `preferences` spaces are missing. Accepting the user agreement and restarting Ghidra fixes the problem as `preferences` then contains `USER_AGREEMENT=ACCEPT`.

Not sure if this is something the Ghidra devs care to investigate further. If alternative window managers are not supported by Ghidra, this issue can be closed.
At least others with the same problem can find the conversation.

**Comment by dragonmacher** [comment]
I investigated briefly the code relating to the User Agreement preference.  I did not see anything special on our side of things.  My guess is that when we show the user agreement, the UI has not been fully initialized, which causes the aforementioned window managers to do something funky.

**Comment by huornlmj** [comment]
I'm seeing this in the v 9.1.2 downloadable from https://ghidra-sre.org/ and running the ./ghidraRun script on Ubuntu 18.04 desktop.  I deleted my ~/.ghidra directory and tried again.  Same issue.

**Comment by kalbfled** [comment]
Ditto what @huornlmj wrote above.  This issue should be reopened.

![ghidra_tips](https://user-images.githubusercontent.com/5375929/75619377-1158e480-5b49-11ea-8f43-686163f192de.png)

**Comment by svidovich** [comment]
Same here. It's usable for sure, just a bit goofy.

**Comment by khae0** [comment]
If it helps - I had the same issue with Ubuntu and the downloaded 9.1.2 public. I did not have openjdk11 installed, so did that and went to start ghidra and experienced the the aforementioned error with missing spaces. I figured it might have to do with user java environment vars and opened a new terminal, which fixed the problem.

**Comment by NilsIrl** [comment]
Same issue on NixOS using Sway and Xwayland. Closing ghidra and restarting it fixed the problem.

Without Xwayland, ghidra doesn't start with the error `Exited with error.  Run in foreground (fg) mode for more details.`.

**Comment by VorpalBlade** [comment]
I'm seeing this on Ubuntu 18.04, using X.Org, with the latest public download from https://ghidra-sre.org (`ghidra_9.1.2_PUBLIC_20200212.zip`)

**Comment by ttepatti** [comment]
Experiencing this issue on a fresh Ubuntu 18.04 with the openjdk-11-jdk JDK package. Ghidra being run with the "ghidraRun" script.

**Comment by NilsIrl** [comment]
I also experienced this on arch linux with X.org and i3. So this has nothing to do with Wayland.

**Comment by ryanmkurtz** [comment]
I did some debugging.  Here are the conditions needed to see the bug:

1. Linux
2. User Agreement pops up
3. Look & Feel is "Nimbus"

Nimbus is the hardcoded default Look & Feel on Linux.  If I change the code to not use Nimbus, I don't see the issue when the User Agreement pops up.  I also made Nimbus the default Look & Feel on Windows hoping that Linux doesn't have anything to do with this issue, but Windows was able to handle Nimbus and the User Agreement fine.
